core/src/main/java/org/elasticsearch/index/get/GetField.java

```diff
@@ -116,9 +116,14 @@ public static GetField fromXContent(XContentParser parser) throws IOException {
         String fieldName = parser.currentName();
         XContentParser.Token token = parser.nextToken();
         ensureExpectedToken(XContentParser.Token.START_ARRAY, token, parser::getTokenLocation);
+
         List<Object> values = new ArrayList<>();
         while((token = parser.nextToken()) != XContentParser.Token.END_ARRAY) {
-            values.add(parseStoredFieldsValue(parser));
+            if (token.isValue()) {
+                values.add(parseStoredFieldsValue(parser));
+            } else {
+                parser.skipChildren();
```

Comment:
maybe this is too much? we do expect each field under fields to contain an array of values I'd say. without this change parseStoredFieldsValue would throw exception if it finds something that it doesn't support, which is probably ok?